### DISCUSSION TITLE
[Debezium] Cleaning up MariaDB specific grammar from MySQL grammar?

### DISCUSSION THREAD

**Discussion by ani-sha** [antlr/grammars-v4#4342] [General]
Hi team,

I am working on aligning the formatting changes done using the `antlr-format-cli` in https://github.com/antlr/grammars-v4/pull/3843 in the Debezium DDL module. While doing so we are also planning to remove MariaDB-specific grammar from MySQL as we now have a separate MariaDB module that has been pulled from the ANTLR4 grammar for MariaDB. 

But I see there is still some MariaDB-specific grammar in the MySQL [grammar](https://github.com/antlr/grammars-v4/tree/master/sql/mysql/Positive-Technologies). Do you have any plans to remove them or clean up the grammar overall? Thanks.

cc: @jpechane

**Comment by kaby76** [comment]
Yes, the https://github.com/antlr/grammars-v4/tree/master/sql/mysql/Positive-Technologies grammar shouldn't contain other sql. It should be only a MySql grammar. So yes,clean up what you can.

The other MySql grammar https://github.com/antlr/grammars-v4/tree/master/sql/mysql/Oracle is the official Oracle grammar modified to be target agnostic. So it makes sense to not modify that any further than what Oracle provides. I am planning to add ports for the rest of the targets to that. Then, the plan is to generate the target agnostic grammar directly from the grammar in the Oracle github repo, which is written for Antlr4ng TypeScript. I'll be starting that work in a few days.

**Comment by ani-sha** [comment]
Hi @kaby76, the CI checks in ANTLR grammar have a check step for formatting the grammar using antlr-format tool but throws a [warning](https://github.com/ani-sha/grammars-v4/blob/451d59c5ef5ba954ffff24afdf2d367497fb9098/_scripts/test-static-checks.sh#L367C1-L377C7) if the grammars are not in the right format. Is that intentional and do you format it periodically rather than formatting it with each PR?

**Reply by kaby76** [reply]
Yes, it's a normal operation.

The script is supposed to only output a warning if the grammar isn't formatted according to the standard. It just means that it may be reworked and reformatted in the future other than yourself.

Intentional or not, some people choose a license for the grammar to be so restrictive that no one can do anything after being submitted, not even fix bugs in the code. Unfortunately, we need to be able to fix bugs in the grammar. People expect grammars to just work.

**Reply by ani-sha** [reply]
Understood. Thanks!
